### DISCUSSION TITLE
Help with self-signed certificate (using internal root CA)

### DISCUSSION THREAD

**Discussion by metemadi** [coder/code-server#5160] [General]
Hi everyone,
Thanks in advance for your help. I am trying to run code server internally at our organization. Our IT provides us with an internal root CA and subordinate CA that is already installed on our clients. Code-server is running in a vanilla ubuntu docker that we have full control over and root access. Looking through the github issues and forums, I performed the following:

1. I got a private key for our server signed by our internal root CA. So I have a .CER for the server as well as the public .CER for the Root and Subordinate CA.
2. Concatenate the public keys of the Root, Subordinate, and Server into one .CER file (all .PEM format)
3. Start code-server with --cert to the above combined.cer file, and --cert-key using the private key of the server (I do not have access to the private key of the CA/subordinate but I don't think that is needed)
4. Setting host to the IP address:443 in the config YAML. Let's say https://internalhost.corp.ourcompany.org

When I visit the server in the browser, I get the certificate common name invalid error... but the certificate itself says internalhost.corp.ourcompany.org in the common name (i.e. it matches perfectly with #4).

I am definitely a bit rusty in all things SSL - any help would be greatly appreciated.

Thank you again!!

**Comment by jsjoeio** [comment]
I am a noob when it comes to things like SSL. Hoping someone from the community can chime in! Otherwise, happy to be a rubber duck!

**Comment by JWleo** [comment]
@metemadi you should use your ip address (such like 21.33.414.121:443) to visit the server in the browser.And the private key is in your server ,the public one is in your PC.
  If you want to visit the server in the browser through internalhost.corp.ourcompany.org,you must use the  internalhost.corp.ourcompany.org to  checkout the domain name resolution of your ip address.
if you still don't know ,you can ask more about this
